extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -28,15 +30,71 @@ class AmpStickyAd extends AMP.BaseElement {
   isLayoutSupported(layout) {
     return layout == Layout.NODISPLAY;
   }
-
-  /** @override */
+  /**
+   * @override
+   * @this {undefined}  // Make linter happy
+   */
   buildCallback() {
     /** @const @private {boolean} */
     this.isExperimentOn_ = isExperimentOn(this.getWin(), TAG);
     if (!this.isExperimentOn_) {
       dev.warn(TAG, `TAG ${TAG} disabled`);
       return;
     }
+    this.viewport_ = this.getViewport();
+    this.isDisplayed_ = false;
+    this.initialScrollTop_ = this.viewport_.getScrollTop();
```

Comment:
I am indeed confused about the code here. I thought sticky-ad should be built anyway and all these measurements should be done after `buildCallback`. I tried putting the code to `layoutCallback` but they never get called as far as I can see. For `scrollHeight` I do need to update its value, maybe I can check it's value on every `onScroll` if that's not too expensive? Or wait for build completion before handle `onScroll` event?
wonder where would be a proper place to handle `onScroll` and schedule layout for sticky-ad.